components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -157,14 +169,114 @@ public void setUp() throws Exception {
+        user2.setUserStoreDomain(TestConstants.USER_STORE_DOMAIN);
+        user2.setTenantDomain(SUPER_TENANT_DOMAIN_NAME);
+        user2.setFederatedUser(true);
+        messageContext2.setAuthorizedUser(user2);
+
+        return new Object[][]{
+                {messageContext, tokenRespDTO, TestConstants.CLIENT_ID},
+                {messageContext2, tokenRespDTO2, "dabfba9390aa423f8b04332794d83614"}
+        };
+    }
+
+    @Test(dataProvider = "TestBuildRequestIDToken")
+    public void testBuildIDToken(OAuthTokenReqMessageContext messageContext,
+                                 OAuth2AccessTokenRespDTO tokenRespDTO, String clientId) throws Exception {
+
+        RealmService realmService = IdentityTenantUtil.getRealmService();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .setUserRealm(realmService.getTenantUserRealm(SUPER_TENANT_ID));
+        IdpMgtServiceComponentHolder.getInstance().setRealmService(IdentityTenantUtil.getRealmService());
+        String idToken = defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getAudience().get(0),
+                clientId);
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getIssuer(),
+                "https://localhost:9443/oauth2/token");
+        }
+
+    @DataProvider(name = "TestBuildAuthzIDToken")
+    public Object[][] buildAuthzIDToken() throws Exception {
+        OAuth2AuthorizeReqDTO authzTokenReqDTO = new OAuth2AuthorizeReqDTO();
+        OAuthAuthzReqMessageContext authzMessageContext = new OAuthAuthzReqMessageContext(authzTokenReqDTO);
+        OAuth2AuthorizeRespDTO authzTokenRespDTO = new OAuth2AuthorizeRespDTO();
+        authzTokenReqDTO.setTenantDomain(SUPER_TENANT_DOMAIN_NAME);
+        authzTokenReqDTO.setConsumerKey(TestConstants.CLIENT_ID);
+        authzTokenReqDTO.setIdpSessionIdentifier(TestConstants.IDP_ENTITY_ID_ALIAS);
+        authzTokenReqDTO.setUser(user);
+        authzTokenRespDTO.setAccessToken(TestConstants.ACCESS_TOKEN);
+
+        OAuth2AuthorizeReqDTO authzTokenReqDTO2 = new OAuth2AuthorizeReqDTO();
+        OAuthAuthzReqMessageContext authzMessageContext2 = new OAuthAuthzReqMessageContext(authzTokenReqDTO2);
+        OAuth2AuthorizeRespDTO authzTokenRespDTO2 = new OAuth2AuthorizeRespDTO();
+        authzTokenReqDTO2.setTenantDomain(SUPER_TENANT_DOMAIN_NAME);
+        authzTokenReqDTO2.setConsumerKey("dabfba9390aa423f8b04332794d83614");
+        authzTokenReqDTO2.setIdpSessionIdentifier(TestConstants.IDP_ENTITY_ID_ALIAS);
+        AuthenticatedUser user2 = new AuthenticatedUser();
+        authzTokenReqDTO2.setUser(user2);
+        user2.setAuthenticatedSubjectIdentifier("user2");
+        user2.setUserName("user2");
+        user2.setUserStoreDomain(TestConstants.USER_STORE_DOMAIN);
+        user2.setTenantDomain(TestConstants.TENANT_DOMAIN);
+        user2.setFederatedUser(true);
```

Comment:
shouldn't this performed before https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1582/files#diff-6911f932c09c723653eab17456f782fd0bc3b2752e7d270cc94ab14c221879b1R233 ?

Reply:
Addressed with 1c087f83ec0cada23b770b13bc32829d7e4ef83e